src/platform/plugins/shared/dashboard/server/content_management/v3/transforms/out/control_state.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { defaults, flow } from 'lodash';
+import { SerializableRecord } from '@kbn/utility-types';
+import { DEFAULT_CONTROL_GROW, DEFAULT_CONTROL_WIDTH } from '@kbn/controls-plugin/common';
+import { ControlGroupAttributes } from '../../types';
+
+/**
+ * Transform functions for serialized controls state.
+ */
+export const transformControlsState: (
+  serializedControlState: string
+) => ControlGroupAttributes['controls'] = flow(
+  JSON.parse,
+  transformControlObjectToArray,
+  transformControlsWidthAuto,
+  transformControlsSetDefaults,
+  transformControlProperties
+);
+
+export function transformControlObjectToArray(controls: Record<string, SerializableRecord>) {
+  return Object.entries(controls).map(([id, control]) => ({ id, ...control }));
+}
+
+/**
+ * Some controls were serialized with width set to 'auto'. This function will transform those controls
+ * to have the default width and grow set to true. See @link https://github.com/elastic/kibana/issues/211113.
+ */
+export function transformControlsWidthAuto(controls: SerializableRecord[]) {
+  return controls.map((control) => {
+    if (control.width === 'auto') {
+      return { ...control, width: DEFAULT_CONTROL_WIDTH, grow: true };
+    }
+    return control;
+  });
+}
+
+// TODO We may want to remove setting defaults in the future
+export function transformControlsSetDefaults(controls: SerializableRecord[]) {
+  return controls.map((control) =>
+    defaults(control, { grow: DEFAULT_CONTROL_GROW, width: DEFAULT_CONTROL_WIDTH })
```

Comment:
I wonder why the usage of lodash `defaults` here rather than a spread? Is it just for understandability?

Reply:
Good old fashioned laziness. But I think it is better to be consistent and spread works just as well without needing to use lodash `defaults`. Updated in https://github.com/elastic/kibana/pull/213831/commits/1f430b0cb256eebea1fd0a1a577978a1fd9642b9.